src/Groups/types.jl

```diff
@@ -0,0 +1,233 @@
+import Hecke:
+    abelian_group,
+    automorphism_group,
+    center,
+    codomain,
+    cokernel,
+    compose,
+    degree,
+    derived_series,
+    direct_product,
+    domain,
+    elem_type,
+    elements,
+    free_abelian_group,
+    gens,
+    haspreimage,
+    hom,
+    id_hom,
+    image,
+    index,
+    inv!,
+    isabelian,
+    isbijective,
+    ischaracteristic,
+    isconjugate,
+    iscyclic,
+    isinjective,
+    isinvertible,
+    isisomorphic,
+    isnormal,
+    issimple,
+    issurjective,
+    kernel,
+    mul,
+    mul!,
+    ngens,
+    one!,
+    order,
+    perm,
+    preimage,
+    quo,
+    representative,
+    small_group,
+    sub,
+    subgroups
+
+import Base: ==, parent, show
+
+import GAP.GapObj
+
+export PermGroup, PermGroupElem
+export MatrixGroup, MatrixGroupElem
+export PcGroup, PcGroupElem
+export FPGroup, FPGroupElem
+export AutomorphismGroup, AutomorphismGroupElem
+export elem_type
+
+"""
+TODO: document this
+"""
+abstract type Group <: AbstractAlgebra.Group end
+
+"""
+TODO: document this
+"""
+struct GAPGroupElem{T<:Group} <: AbstractAlgebra.GroupElem
+   parent::T
+   X::GapObj
```

Comment:
I think for this, but also for `GroupCoset`, `PermGroup`, etc., there should be a uniform way to access the underlying GAP object, as a clean "escape hatch" for cases where some GAP function is not (yet) wrapped by us.

Of course right now, one can just write `myGroup.X` to do that. But (a) the resulting code is not very readable, and (b) that ties the code to an implementation detail we might want to change eventually? So perhaps an `underlying_gapobj` function with methods for all relevant types should be added? Suggestions for a better name welcome (`extrep` would be the GAP-like name; `unwrap`? ... ?)